core-bundle/src/Routing/Content/StringUrl.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\Content;
+
+final class StringUrl
+{
+    public function __construct(public readonly string $value)
```

Comment:
What if `$value` is not a URL?

Reply:
You mean what if it is an insert tag? The `StringUrl` object will be handled recursively by the `StringUrlResolver`, which will replace insert tags and make the URL absolute.

Reply:
`StringUrl` must accept an Insert Tag as its `$value` as this will then replaced by `StringResolver`. And a string like `foo` would also be a valid (directory relative) URL, technically.

Reply:
> There is no need to check

Ok, so what happens if I pass `"\0"` as argument?

Reply:
A test would tell you. If `if (!$value)` is not true, then it will probably return that value in the end. If you want to enforce a URL, we could check that the URL is absolute after replacing insert tags and applying UrlHelper.

Reply:
That was my initial question. Should we check that `$value` is actually a URL or does it not matter?

Reply:
We actually do 😎 
https://github.com/contao/contao/pull/6596/files#diff-984c96268523339467edac8695ebd6939603adfe8418c38e32f17e37f85152ceR25-R27

Reply:
These lines belong to the `ContentUrlResult` class. We are talking about the `StringUrl` class here.

Reply:
> Should we check that `$value` is actually a URL or does it not matter?

I think it does not matter here. If we later decide that a check is needed it has to be done in the `StringResolver` service, as only this service knows which strings are valid (possibly extended by an extension for example).